.github/workflows/rubocop.yml

```diff
@@ -43,23 +34,46 @@ jobs:
         run: ruby -I lib -r bundler/setup -r rubocop -e 'exit 0'
       - name: Set up Coverage
         # Only collect coverage data on ubuntu runners with CRuby
-        if: matrix.os == 'ubuntu' && !startsWith(matrix.ruby, 'jruby-')
+        if: ${{ !startsWith(matrix.ruby, 'jruby-') }}
         run: echo "COVERAGE=true" >> $GITHUB_ENV
       - name: spec
         env:
           CI_RUBY_VERSION: ${{ matrix.ruby }}
         run: bundle exec rake spec
       - name: Upload Coverage Artifact
-        if: matrix.os == 'ubuntu' && !startsWith(matrix.ruby, 'jruby-')
+        if: ${{ !startsWith(matrix.ruby, 'jruby-') }}
         uses: actions/upload-artifact@v4
         with:
-          name: coverage-${{ matrix.os }}-${{ matrix.ruby }}
+          name: coverage-ubuntu-${{ matrix.ruby }}
           path: coverage/.resultset.json
           if-no-files-found: error
 
+  spec-windows:
+    needs: spec-ubuntu
+    name: Spec - windows ${{ matrix.ruby }}
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ['2.7', '3.0', '3.1', '3.2', '3.3', 'mingw']
```

Comment:
I'm also wondering if we need to run all the Ruby versions here or we can again just go for the oldest/newest combo. I'm struggling to come up with a scenario where on Windows we'd have failures that are both Windows-specific and specific to a specific Ruby version.

Reply:
I agree, and I’ve reduced the matrix to just 2.7, 3.3 and mingw.

I am not sure what mingw is and whether we need it.

Reply:
That seems to be a differently build version of ruby. Just specifying a version pulls the `ucrt` variant. Considering that RuboCop doesn't do C extensions I fail to see the value in testing against that.

Reply:
I will remove `mingw` then.